components-registry-service-core/src/main/kotlin/org/octopusden/octopus/components/registry/core/dto/ToolDTO.kt

```diff
@@ -0,0 +1,14 @@
+package org.octopusden.octopus.components.registry.core.dto
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+data class ToolDTO @JsonCreator constructor(
+    @JsonProperty("name") val name: String? = null,
```

Comment:
can we make this properties as non-nulable?

Reply:
Yes, it is possible to make the name property non-nullable, as the corresponding `Tool` data class in the `component-resolver-api` already defines it as non-nullable. However, I would need confirmation on whether every Tool instance is guaranteed to have a name. If so, I can safely enforce non-nullability here.

Reply:
@adellinekania , you need to have guarantee via validation on EscrowConfig loading

@pgorbachev , what about another fields? Do we have a contract?

Reply:
@adellinekania , 
All the parameters are mandatory except install script.
So 
1) Add validateBuildParameters to EscrowConfigValidator.groovy to ensure this (add test for this in EscrowConfigurationLoaderTest)
2) make DTO fields non-null except install script

Reply:
@adellinekania , we talked about nonNullable fields and fields still nullable